101-data-factory-blob-to-sql-copy/azuredeploy.json

```diff
@@ -1,89 +1,94 @@
     "parameters": {
       "dataFactoryName": {
         "type": "string",
         "metadata": {
-          "description":  "Data factory name"
+          "description": "Data factory name"
         }
       },
       "storageAccountName": {
         "type": "string",
         "metadata": {
-          "description":  "Azure Storage account name"
+          "description": "Name of the Azure storage account that contains the data to be copied."
         }
       },
       "storageAccountKey": {
         "type": "securestring",
         "metadata": {
-          "description":  "Azure Storage account key"
+          "description": "Key for the Azure storage account."
         }
       },
       "sourceBlobContainer": {
         "type": "string",
         "metadata": {
-          "description":  "Azure blob container name"
+          "description": "Name of the blob container in the Azure Storage account."
         }
       },
       "sourceBlobName": {
         "type": "string",
         "metadata": {
-          "description":  "Azure blob name"
+          "description": "Name of the blob in the container that has the data to be copied to Azure SQL Database table"
         }
       },
       "sqlServerName": {
         "type": "string",
         "metadata": {
-          "description":  "Azure SQL server name"
+          "description": "Name of the Azure SQL Server that will hold the output/copied data."
         }
       },
       "databaseName": {
         "type": "string",
         "metadata": {
-          "description":  "Azure SQL database name"
+          "description": "Name of the Azure SQL Database in the Azure SQL server."
         }
       },
       "sqlServerUserName": {
         "type": "string",
         "metadata": {
-          "description":  "Azure SQL server - user name"
+          "description": "Name of the user that has access to the Azure SQL server."
         }
       },
       "sqlServerPassword": {
         "type": "securestring",
         "metadata": {
-          "description":  "Azure SQL server - user password"
+          "description": "Password for the user."
         }
       },
       "targetSQLTable": {
         "type": "string",
         "metadata": {
-          "description":  "Azure SQL table name"
-        } 
+          "description": "Table in the Azure SQL Database that will hold the copied data."
+        }
       }
     },
     "variables": {
-      "apiVersion": "2015-10-01",
-      "storageApiVersion" :  "2015-05-01-preview",
       "azureSqlLinkedServiceName": "AzureSqlLinkedService",
       "azureStorageLinkedServiceName": "AzureStorageLinkedService",
       "blobInputDatasetName": "BlobInputDataset",
       "sqlOutputDatasetName": "SQLOutputDataset",
-      "pipelineName": "Blob2SQLPipeline"    
+      "pipelineName": "Blob2SQLPipeline",
+      "dataFactoryID": "[resourceId('Microsoft.DataFactory/dataFactories/', parameters('dataFactoryName'))]",
```

Comment:
One simplification you can make here... if these are only used for dependsOn (looks like they are) then you can simply use the name of the resource since it's defined in the template. e.g. 

"dependsOn": [ parameters('dataFactoryName') ]

The full resourceId is only needed if it's not defined in this template (e.g. a nested template) or you have resources of different types with the same name...